## Project 2 - Programming for Data Analysis

### Course: HDip in Computing in Data Analytics
### Module: Programming for Data Analysis
### Student: Sarah Hastings
### Student Number: G00235562
***

## An analysis of paleo-present climate data


• Analyse CO2 vs Temperature Anomaly from 800kyrs – present.

• Examine one other (paleo/modern) features (e.g. CH4 or polar ice-coverage)

• Examine Irish context:
o Climate change signals: (see Maynooth study: The emergence of a climate change 
signal in long-term Irish meteorological observations - ScienceDirect)

• Fuse and analyse data from various data sources and format fused data set as a pandas 
dataframe and export to csv and json formats

• For all of the above variables, analyse the data, the trends and the relationships between 
them (temporal leads/lags/frequency analysis).

• Predict global temperature anomaly over next few decades (synthesise data) and compare to 
published climate models if atmospheric CO2 trends continue

• Comment on accelerated warming based on very latest features (e.g. temperature/polar-icecoverag

### Importing libraries

In [3]:
# Import pandas to allow for data analysis, manipulation, work with dataframes
import pandas as pd
# Use Pyplot, part of the Matplotlib library to visualize diagrams/use for visual data/plots etc 
import matplotlib.pyplot as plt
# Import numpy to work with numerical data, arrays
import numpy as np
# Importing seaborn gives a variety of visualization patterns in addition to matplotlib
import seaborn as sns
# Useful for plotting visual relationships between data 
from pandas.plotting import andrews_curves
# Split the dataset into train and test models
from sklearn.model_selection import train_test_split
# Training model
from sklearn.preprocessing import StandardScaler
# Classification and Logistic regression
from sklearn.linear_model import LogisticRegression
# View accuracy/predictability
from sklearn.metrics import accuracy_score,confusion_matrix

In [ ]:
https://council.science/current/blog/climate-explained-why-is-the-arctic-warming-faster-than-other-parts-of-the-world/

### 1. Analyse CO2 vs Temperature Anomaly from 800kyrs – present

what doing, what aim 
trends, patterns, correlations, etc on erths response to changing co2 

 how changes in atmospheric carbon dioxide levels correspond to variations in temperature anomalies. This relationship is crucial in understanding the impact of greenhouse gas emissions on the Earth's climate
 
 Positive Relationship: An increase in CO2 levels is often associated with a positive temperature anomaly, indicating a warming trend. This aligns with the concept of the greenhouse effect, where higher concentrations of greenhouse gases trap more heat in the atmosphere.

Negative Relationship: While the overall trend is positive, variations may exist due to other factors influencing climate, such as natural climate variability, solar radiation, or volcanic activity.

 exploration of the correlation between atmospheric CO2 levels and variations in temperature anomalies over time, providing insights into the impact of human-induced greenhouse gas emissions on global climate pattern
 
 
 
 

### Lüthi, D et al. (2008) - High-resolution carbon dioxide concentration record 650,000–800,000 years before present

In [19]:
# Specify the path to your Excel file
file_path = './data/41586_2008_BFnature06949_MOESM31_ESM.xls'

# Read the specified sheet into a Pandas DataFrame
luthi = pd.read_excel(file_path, sheet_name='3.  Composite CO2', skiprows=6, index_col=None)
luthi

EDC3_gas_a (yr)  CO2 (ppmv)
0                 137       280.4
1                 268       274.9
2                 279       277.9
3                 395       279.1
4                 404       281.9
...               ...         ...
1091           794608       199.4
1092           795202       195.2
1093           796467       189.3
1094           797099       188.4
1095           798512       191.0

[1096 rows x 2 columns]

### IPCC Data - C02 Composite Antartic Ice Core

In [15]:
# Specify the path to your Excel file
file_path = './data/grl52461-sup-0003-supplementary.xls'

# Read the specified sheet into a Pandas DataFrame
ipcc = pd.read_excel(file_path, sheet_name='CO2 Composite', skiprows=14, index_col=None)
ipcc

Gasage (yr BP)   CO2 (ppmv)  sigma mean CO2 (ppmv)
0          -51.030000  368.022488               0.060442
1          -48.000000  361.780737               0.370000
2          -46.279272  359.647793               0.098000
3          -44.405642  357.106740               0.159923
4          -43.080000  353.946685               0.043007
...               ...         ...                    ...
1896    803925.284376  202.921723               2.064488
1897    804009.870607  207.498645               0.915083
1898    804522.674630  204.861938               1.642851
1899    805132.442334  202.226839               0.689587
1900    805668.868405  207.285440               2.202808

[1901 rows x 3 columns]

### Temperature

### EPICA Dome C Ice Core 800KYr deuterium data and temperature estimates. PANGAEA

In [16]:
# Specify the path to your Excel file
file_path = './data/epicaDC.deuttemp.EDC3-AICC.xlsx'

# Read the specified sheet into a Pandas DataFrame
jouzel = pd.read_excel(file_path)
jouzel

bag     ztop      EDC3béta       AICC2012  deutfinal   temp  \
0        1     0.00     -50.00000     -55.000000        NaN  218.3   
1        2     0.55     -43.55977     -50.513333        NaN  218.3   
2        3     1.10     -37.44019     -46.026667        NaN  218.3   
3        4     1.65     -31.64156     -41.540000        NaN  218.3   
4        5     2.20     -24.55278     -34.516667        NaN  218.3   
...    ...      ...           ...            ...        ...    ...   
5814  5815  3197.70  816871.60000  817562.000000     -432.2  211.0   
5815  5816  3198.25  817693.60000  818631.000000     -431.4  211.2   
5816  5817  3198.80  818505.50000  819700.000000     -430.3  211.3   
5817  5818  3199.35  819303.10000  820769.000000     -429.8  211.4   
5818  5819  3199.90  820093.90000  821838.000000     -430.0  211.4   

      acc-EDC3beta  
0         3.105321  
1         3.104805  
2         3.104404  
3         3.104025  
4         3.103453  
...            ...  
5814      1.529000  
5815      1.548000  
5816      1.576000  
5817      1.589000  
5818      1.583000  

[5819 rows x 7 columns]

### Methane

In [24]:
# Specify the path to your Excel file
file_path = './data/ghg-concentrations_fig-2.csv'

# Read the specified sheet into a Pandas DataFrame
methane = pd.read_csv(file_path, skiprows=6)
methane

Year (negative values = BC)  EPICA Dome C, Antarctica Law Dome  \
0          Ice core measurements                       NaN      NaN   
1                      -797446.1                     399.3      NaN   
2                      -796467.4                     457.8      NaN   
3                        -795327                     396.1      NaN   
4                      -794369.7                     418.0      NaN   
...                          ...                       ...      ...   
2210                        2017                       NaN      NaN   
2211                        2018                       NaN      NaN   
2212                        2019                       NaN      NaN   
2213                        2020                       NaN      NaN   
2214                        2021                       NaN      NaN   

     Cape Grim, Australia Mauna Loa, Hawaii Shetland Islands, Scotland  
0                     NaN               NaN                        NaN  
1                     NaN               NaN                        NaN  
2                     NaN               NaN                        NaN  
3                     NaN               NaN                        NaN  
4                     NaN               NaN                        NaN  
...                   ...               ...                        ...  
2210             1,798.62          1,865.50                        NaN  
2211             1,810.81          1,871.53                        NaN  
2212             1,816.61          1,882.55                        NaN  
2213             1,824.70          1,891.47                        NaN  
2214             1,841.53               NaN                        NaN  

[2215 rows x 6 columns]

### Irish Context

https://www.climate.gov/maps-data/dataset/natural-records-past-climate-maps-visualizations-and-descriptive-information


Paleoclimatology is the study of previous climates that have existed during Earth's different geologic ages. Paleoclimatologists try to identify the causes of climate changes that have happened in the past in order to better understand our present and future climate

Paleoclimatology data are derived from natural sources such as tree rings, cave deposits, ice cores, fire history, coral growth bands, pollen, stalagmites, and ocean and lake sediments. These proxy climate data extend the weather and climate information archive by hundreds to millions of years. This is the world's largest archive of paleoclimatology data, with contributions from the international scientific community. The data include geophysical or biological measurement time series and some reconstructed climate variables such as temperature and precipitation.
    
The time span "800,000 years to present" refers to a geological and climatological timeframe. In this context, it likely pertains to the study of paleoclimate data, which involves analyzing climate conditions, atmospheric composition, and other environmental factors over an extended period.

When researchers refer to "800,000 years to present," they are typically looking at various proxies and records, such as ice cores, sediment layers, tree rings, and other indicators, to reconstruct and understand Earth's climate history. This extensive time frame allows scientists to investigate long-term climate trends, natural variations, and the impact of different factors on the Earth's climate system.

In the context of the previous question regarding the analysis of CO2 versus temperature anomaly over this period, it suggests a comprehensive study of the interplay between greenhouse gas concentrations and temperature variations throughout a significant portion of Earth's history.

Paleoclimatology is the study of the climate history of Earth. This science helps people better understand the climate of Earth in the past and how it relates to the present and future climate on the planet
Since modern records do not outline most of Earth’s climatic past, scientists must gather data preserved in nature over the millennia in paleological remains referred to as proxy records.

Scientists can also drill into ice sheets and extract ice cores to analyze deposits trapped within the ice, such as pollen and gas. This allows them to better understand atmospheric and climatic conditions that existed when particular layers of the sheet formed. All of this proxy data is synthesized to develop an understanding of Earth’s climate over different geologic periods.

The field of paleoclimatology has taught scientists that the climate of Earth is in a perpetual state of change. Research has revealed that there have been several ice ages as well as several periods of global warming in Earth's history. Additionally, scientists have determined that there can be both short- and long-term trends in climate. For instance, while climate change can occur—and often has occurred—over an extended amount of time, paleoclimatologists have also recorded shorter warming periods during an ice age. Scientists have become aware that Earth is very dynamic and that past events might help us better understand the past and prepare for future climate changes.

Paleoclimatology has also helped scientists study and understand how other environmental factors, such as continental drift, solar energy, greenhouses gases in the atmosphere, and the variation in Earth’s orbit have all affected the climate of Earth over time. The science of paleoclimatology is vital to our understanding of climate on Earth. As scientists become increasingly aware of how climates have been influenced in the past, they can develop models that help predict how increased carbon dioxide levels and other changes might impact the climate of Earth in the future.

https://www.universityofgalway.ie/pru/

https://www.epa.ie/publications/research/climate-change/Reserach_Report_277.pdf

Paleoclimatology is the study of past climates. Since it is not possible to go back in time to see what climates were like, scientists use imprints created during past climate, known as proxies, to interpret paleoclimate. Organisms, such as diatoms, forams, and coral serve as useful climate proxies. Other proxies include ice cores, tree rings, and sediment cores (which include diatoms, foraminifera, microbiota, pollen, and charcoal within the sediment and the sediment itself).

Past climate can be reconstructed using a combination of different types of proxy records. These records can then be integrated with observations of Earth's modern climate and placed into a computer model to infer past as well as predict future climate.

In [ ]:
Paleoclimatology is the study of past climates. Since it is not possible to go back in time to see what climates were like, scientists use imprints created during past climate, known as proxies, to interpret paleoclimate. Organisms, such as diatoms, forams, and coral serve as useful climate proxies. Other proxies include ice cores, tree rings, and sediment cores (which include diatoms, foraminifera, microbiota, pollen, and charcoal within the sediment and the sediment itself).

Past climate can be reconstructed using a combination of different types of proxy records. These records can then be integrated with observations of Earth's modern climate and placed into a computer model to infer past as well as predict future climate.

The science of paleoclimatology is important for past, contemporary, and future issues. Understanding past climate helps us to explain how current ecosystems came to be. For example, climate typically controls what types of vegetation grow in a particular area. Furthermore, paleoclimatology provides data that we can use to model and predict both current and future climate change scenarios. Computer models can be used to study the potential effect of increased atmospheric carbon dioxide on climate.

With a system as complex as Earth's climate, it is a daunting task for scientists to be able to make projections about future climate changes and how it may affect the distribution of plants and animals. However, paleoclimate data are used as a foundation for climate scientists by providing crucial information such as rates of past climate change and how vegetation and animal populations responded to the change. Computer models can be used predict different future climate patterns, and paleoclimate data provides a useful framework from which to base these models. For more information about climate proxies and climate change, see the collections of web-based materials below

### An analysis of irish Context

https://www.dias.ie/wp-content/uploads/2016/07/Session2_VRath_Iretherm2016.pdf


https://www.newyorkfed.org/medialibrary/media/research/staff_reports/sr1031.pdf

https://www.ncei.noaa.gov/products/paleoclimatology/paleo-perspectives/global-warming

https://www.climate.gov/news-features/understanding-climate/climate-change-atmospheric-carbon-dioxide#:~:text=Based%20on%20air%20bubbles%20trapped,was%20280%20ppm%20or%20less.

https://www.climate.gov/news-features/understanding-climate/climate-change-atmospheric-carbon-dioxide


Based on air bubbles trapped in mile-thick ice cores and other paleoclimate evidence, we know that during the ice age cycles of the past million years or so, atmospheric carbon dioxide never exceeded 300 ppm. Before the Industrial Revolution started in the mid-1700s, atmospheric carbon dioxide was 280 ppm or less

### References

https://education.nationalgeographic.org/resource/paleoclimatology-RL/

https://www.ipcc.ch/site/assets/uploads/2018/02/WG1AR5_Chapter05_FINAL.pdf

https://www.ncei.noaa.gov/access/paleo-search/study/6080

https://www.met.ie/climate/available-data/long-term-data-sets/

https://www.sciencedirect.com/science/article/pii/S2212094723000610#bib13

https://edition.cnn.com/2023/11/20/climate/2-degree-warming-limit-record-copernicus-climate-int/index.html